sdk/go/common/resource/properties_path.go

```diff
@@ -257,3 +257,37 @@ func (p PropertyPath) Delete(dest PropertyValue) bool {
 	return true
 
 }
+
+// Contains returns true if the receiver property path contains the other pproperty path.
+// For example, the path `foo["bar"][1]` contains the path `foo.bar[1].baz`.  The key `"*"`
+// is a wildcard which matches any string or int index at that same nesting level.  So for example,
+// the path `foo.*.baz` contains `foo.bar.baz.bam`, and the path `*` contains any path.
```

Comment:
This comment doesn't seem accurate w.r.t the code. From what I see, `*` only works with dotted property accesses, not with brackets.

Reply:
Good catch - I suggest we update the code to match the comment, and add a test covering this.

Reply:
I added some more test cases for wildcards, and confirmed that the existing logic works as expected.

Reply:
@lblackstone are you also going to update the code to match the comment, or will we take care of that in a future change?

Reply:
Oh - indeed as @lblackstone noted above - the code does actually match the comment. The new tests make this clear.

Reply:
Ah, I see it now! Tricky. Thanks for pointing that out.